src/FSharpLint.Core/Rules/Conventions/FriendlyAsyncOverload.fs

```diff
@@ -0,0 +1,71 @@
+module FSharpLint.Rules.FriendlyAsyncOverload
+
+open FSharpLint.Framework
+open FSharpLint.Framework.Suggestion
+open FSharp.Compiler.Syntax
+open FSharp.Compiler.Text
+open FSharpLint.Framework.Ast
+open FSharpLint.Framework.Rules
+open System
+
+type NodeDetails = { Ident: string; Range: range }
+
+let rec private getIdentFromSynPat =
+    function
+    | SynPat.LongIdent (longDotId = longDotId) ->
+        longDotId
+        |> ExpressionUtilities.longIdentWithDotsToString
+        |> Some
+    | SynPat.Typed (pat, _, _) -> getIdentFromSynPat pat
+    | _ -> None
+
+let runner (args: AstNodeRuleParams) =
+    let hasAsync (syntaxArray: AbstractSyntaxArray.Node []) nodeIndex fnIdent =
+        let rec hasAsync index =
+            if index >= syntaxArray.Length then
+                None
+            else
+                let node = syntaxArray.[index].Actual
+                match node with
+                | AstNode.Binding (SynBinding (_, _, _, _, _attributes, _, _, pattern, _, _, range, _)) ->
+                    match getIdentFromSynPat pattern with
+                    | Some ident when ident = fnIdent + "Async" ->
+                        Some(
+                            { Ident = fnIdent
+                              Range = range }
+                        )
+                    | _ -> hasAsync (index + 1)
+                | _ -> hasAsync (index + 1)
+
+        hasAsync nodeIndex
+
+    match args.AstNode with
+    | AstNode.Binding (SynBinding (_, _, _, _, _, _, _, pattern, synInfo, _, range, _)) ->
+        match synInfo with
+        | Some (SynBindingReturnInfo ((SynType.App(SynType.LongIdent(LongIdentWithDots(ident,_)),_,_,_,_,_,_)),_,_)) ->
+            match ident with
+            | head::_ when head.idText = "Async" ->
+                let idents = getIdentFromSynPat pattern
+                match idents with
+                | Some ident when ident.EndsWith("Async") = false ->
```

Comment:
no need for parenthesis here (and let's use the operator `not` instead of `= false`)